### PR TITLE
rofi: add support for lists in rasi

### PR DESCRIPTION
**Pull request by thiagokokada**
### Description

Just a small thing that I missed from #1748. This is now supported without the need of using strings to represent them:
https://github.com/davatorium/rofi-themes/blob/9a843caa86d15038fda5455b7fb951569dbd227a/Official%20Themes/solarized.rasi#L134

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [ ] Code tested through `nix-shell --pure tests -A run.all`.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/doc/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [ ] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [ ] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

**Comment by thiagokokada** [comment]
CC @rycee .

File: modules/programs/rofi.nix [hunk 1]
```diff
@@ -331,20 +335,27 @@ in {
       default = null;
       type = with types; nullOr (oneOf [ str path themeType ]);
       example = literalExample ''
-        with config.lib.formats.rasi; {
+        let
+          # config.lib.formats.rasi.mkLiteral unquotes a string, i.e.:
+          # l "abc" -> abc
+          l = config.lib.formats.rasi.mkLiteral;
```
Comment (berbiche) [inline modules/programs/rofi.nix:341]: I'd prefer `inherit (config.lib.formats.rasi) mkLiteral;` instead of using a cryptic nameless function.

**Review by berbiche — CHANGES_REQUESTED** [review]
Thanks for the contribution.

One small change and it's ready to be merged :)

File: modules/programs/rofi.nix (hunk 1 above)
Comment (thiagokokada) [reply modules/programs/rofi.nix]: Done.

**Pull request merged**